### ISSUE TITLE
Hardcore Almost Fixed - But Not Yet

### ISSUE THREAD

**Issue by 1trueway** [dmustanger/7dtd-ServerTools#409]
It's only deleting the players' ".map" file, and it needs to delete the ".ttp" and the ".ttp.bak" files. Once it does this, the player will actually restart. As it is now, it boots you from server after respawning from the last death, and on relog you come back same level with all your items still there.

**Comment by ObsComp** [comment]
Odd. I fixed that with a delayed timer since it writes it one last time after you leave the server. I guess I could try adding more time on the timer. Which version are you running?

**Comment by 1trueway** [comment]
server tools 20.6.2
7dtd 20.6 (b9)

is it deleting all 3 of the player files when you test it? maybe it's something to do with me running it on ping perfect

**Comment by 1trueway** [comment]
Im at work and can't check, but its the latest version of server tools as
of date of submission of this issue, and im running the current stable
build of 7dtd.

Not sure how server tools goes about deleting files. If it works for you,
maybe it's a problem with how I've set it up with my server on ping
perfect. It is deleting the .map, just not the ".ttp" and the ".ttp.bak".
It would seem that if it can manage to delete one file, it shouldn't have
issue deleting the other 2.

**Comment by ObsComp** [comment]
It is an issue with timing. ST was running too fast and would attempt to delete the user profile file which is the .tpp file you mention.

It was either in use when attempting to delete it or possibly deleted it, but the game would still write the file one last time and so it was still there when the player rejoined which was then loaded up on the client side.

I have added a timer to the file removal and a blocker that will stop the player from entering the server until the timer expires. I have tested it a few times and it cleared up without issue.

When they meet the criteria for Hardcore death count, it will show their stats in chat like normal, wait 20 seconds to kick them, then after 5 seconds it will delete the file and then allow them back in. I also updated the data it clears up to match the most current. It will be fixed for next release.

Regards,
OC

________________________________
